spec.md

```diff
@@ -388,13 +388,34 @@ both appear within the same JSON object.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
+### application-properties
+* Type: `Map`
+* Description: A list of "key-value" pairs providing information about the
+  event, which are used by an event-based application. The "key-value" pairs
+  MUST be in a flat structure. The value MUST be a string. There MUST be no
+  duplication of keys. One usage example: an application workflow can use a
+  "key-value" pair to correlate multiple events (from different event sources)
+  associated with a serverless application workflow.
+* Constraints:
+  * REQUIRED
+* Examples:
+``` JSON
+  application-properties
+  {
+     "building address": "12 Main Street, Copenhagen, Denmark, 123456",
+     "floor": "2",
+     "apartmentId": "345",
+     "sensor-location": "room1.window2"
```

Comment:
If we adopt @clemensv's proposal for serializing maps then the HTTP header for this would be:
`CE-application-properties-sensor-location` - seems kind of long. And yes I'm going to revisit my "white whale" here.... :-)  I don't see why `CE-application-properties-sensor-location` would be easier to parse or understand than just `CE-sensor-location`. Nor does the description text here explain why I would use this property instead of an extension, in fact, given the length of these names I would opt for the shorter one every time just to not be annoyed. There is nothing that would prevent this property from being an extension and work the exact same way, so we're giving people two choices for almost the same thing (w/o a clear reason to use each), and I think that's going to lead to confusion and interop isuses when the receiver expects them in one but the sender puts it in the other by mistake.